File: crates/normalize-languages/tests/fixtures/cpp/sample.cpp
#include <iostream>
#include <vector>
#include <stdexcept>

template <typename T>
class Stack {
public:
    // Pushes an item onto the stack.
    void push(const T& item) {
        items.push_back(item);
    }

    T pop() {
        if (items.empty()) {
            throw std::underflow_error("Stack is empty");
        }
        T top = items.back();
        items.pop_back();
        return top;
    }

    const T& peek() const {
        if (items.empty()) {
            throw std::underflow_error("Stack is empty");
        }
        return items.back();
    }

    bool empty() const { return items.empty(); }
    std::size_t size() const { return items.size(); }

private:
    std::vector<T> items;
};

std::string classify(int n) {
    if (n < 0) {
        return "negative";
    } else if (n == 0) {
        return "zero";
    } else {
        return "positive";
    }
}

int sum_evens(const std::vector<int>& numbers) {
    int total = 0;
    for (int n : numbers) {
        if (n % 2 == 0) {
            total += n;
        }
    }
    return total;
}

int main() {
    Stack<int> s;
    s.push(10);
    s.push(20);
    std::cout << s.pop() << "\n";
    std::cout << classify(-3) << "\n";
    std::vector<int> nums = {1, 2, 3, 4, 5};
    std::cout << sum_evens(nums) << "\n";
    return 0;
}

File: crates/normalize-languages/tests/fixtures/php/sample.php
<?php

use App\Models\User;
use Illuminate\Support\Collection;

class Stack {
    private array $items = [];

    public function push(mixed $item): void {
        array_push($this->items, $item);
    }

    public function pop(): mixed {
        if (empty($this->items)) {
            throw new \UnderflowException("Stack is empty");
        }
        return array_pop($this->items);
    }

    public function peek(): mixed {
        if (empty($this->items)) {
            return null;
        }
        return end($this->items);
    }

    public function isEmpty(): bool {
        return empty($this->items);
    }

    public function size(): int {
        return count($this->items);
    }
}

/**
 * Classify a number as negative, zero, or positive.
 */
#[Pure]
function classify(int $n): string {
    if ($n < 0) {
        return "negative";
    } elseif ($n === 0) {
        return "zero";
    } else {
        return "positive";
    }
}

function sumEvens(array $numbers): int {
    $total = 0;
    foreach ($numbers as $n) {
        if ($n % 2 === 0) {
            $total += $n;
        }
    }
    return $total;
}

$stack = new Stack();
$stack->push(1);
$stack->push(2);
echo $stack->pop() . "\n";
echo classify(-5) . "\n";
echo sumEvens([1, 2, 3, 4, 5]) . "\n";
